source/content.js

```diff
@@ -24,6 +24,43 @@ function hideListAddActivity() {
 	$('#stream-items-id .js-activity-list_member_added').css('display', 'none');
 }
 
+function inlineInstagramPosts() {
+	$('a.twitter-timeline-link[data-expanded-url*="//www.instagram.com').each((idx, instagramAnchor) => {
+		const tweetElement = $(instagramAnchor).parents('.js-tweet-text-container');
+		// Don't do anything else if the image is already embedded
+		if ($(tweetElement).siblings('.AdaptiveMedia').length > 0) {
+			return true;
+		}
+		const instagramPostUrl = instagramAnchor.dataset.expandedUrl;
+		const imageRegex = /"display_url": ?"([^"]+)"/g;
+		fetchInstagramPhoto(instagramPostUrl)
+			.then(postHtml => {
+				const matches = imageRegex.exec(postHtml);
+				const instagramImageUrl = matches[1] || null;
+				if (instagramImageUrl) {
+					insertInstagramPhotoInto(tweetElement, instagramImageUrl, instagramPostUrl);
+				}
+			});
+	});
+}
+
+function insertInstagramPhotoInto(element, imageUrl, postUrl) {
+	const tweetImageTemplate = `
+	<div class="AdaptiveMedia">
+		<a href="${postUrl}" target="_blank">
+			<img class="refined-instagram-inline" src="${imageUrl}" />
+		</a>
+	</div>
```

Comment:
You can use JSX here. Example: https://github.com/sindresorhus/refined-github/blob/d10510f660d4e1393ad01af097f97de36ced680c/source/features/add-delete-fork-link.js#L17-L19

Just make sure you import `dom-chef` in files you use it: `import {h} from 'dom-chef';`

The benefit of using JSX here is syntax highlighting, performance, and XSS prevention.